Jenkinsfile

```diff
@@ -0,0 +1,259 @@
+
+def config = {
+    owner = 'ksql'
+    slackChannel = '#ksql-alerts'
+    ksql_db_version = "0.6.0-SNAPSHOT"
+    cp_version = "5.5.0-SNAPSHOT" //After done testing this needs to be a beta version.
+    dockerRegistry = '368821881613.dkr.ecr.us-west-2.amazonaws.com/'
+    dockerRepos = ['confluentinc/ksql-cli', 'confluentinc/ksql-rest-app']
+    nodeLabel = 'docker-oraclejdk8-compose-swarm'
+    dockerScan = true
+    cron = '@nightly'
+}
+
+def defaultParams = [
+    booleanParam(name: 'RELEASE_BUILD',
+        description: 'Is this a release build.',
+        defaultValue: false),
+    string(name: 'GIT_REVISION',
+        description: 'The git SHA to create the release build from.',
+        defaultValue: ''),
+    booleanParam(name: 'PROMOTE_TO_PRODUCTION',
+        defaultValue: false,
+        description: 'Promote images to production (DockerHub) for release build images only.'),
+    string(name: 'KSQLDB_VERSION',
+        defaultValue: '',
+        description: 'KSQLDB version to promote to production.'),
+    string(name: 'IMAGE_REVISION',
+        defaultValue: '1',
+        description: 'Revision for Docker images. This is used with PROMOTE_TO_PRODUCTION only'),
+    booleanParam(name: 'UPDATE_LATEST_TAG',
+        defaultValue: false,
+        description: 'Should the latest tag on docker hub be updated to point at this new image version.')
+]
+
+def updateConfig = { c ->
+    c.properties = [parameters(defaultParams)]
+}
+
+def finalConfig = jobConfig(config, [:], updateConfig)
+
+def job = {
+    if (params.RELEASE_BUILD && params.PROMOTE_TO_PRODUCTION) {
+        currentBuild.result = 'ABORTED'
+        error("You can not do a release build and promote to production at the same time.")
+    }
+
+    if (params.RELEASE_BUILD && params.GIT_REVISION == '') {
+        currentBuild.result = 'ABORTED'
+        error("You must provide the GIT_REVISION when doing a release build.")
+    }
+
+    if (params.PROMOTE_TO_PRODUCTION && params.KSQLDB_VERSION == '') {
+        currentBuild.result = 'ABORTED'
+        error("You must provide the KSQLDB_VERSION when promoting images to docker hub.")
+    }
+
+    config.release = params.RELEASE_BUILD
+
+    if (config.release) {
+        // For a release build check out the provided git sha instead of the master branch.
+        if (config.revision == '') {
+            error("If you are doing a release build you must provide a git sha.")
+        }
+        
+        config.revision = params.GIT_REVISION
+        // For a release build we remove the -SNAPSHOT from the version.
+        config.ksql_db_version = config.ksql_db_version.tokenize("-")[0]
+        config.docker_tag = config.ksql_db_version
+    } else {
+        config.docker_tag = config.ksql_db_version.tokenize("-")[0] + '-' + env.BUILD_NUMBER
+        config.revision = 'refs/heads/master'
```

Comment:
I can do this in a follow-up, but I may want to specify a revision that isn't head/master but not release anything. we should allow the param to resolve here too

Reply:
@agavra yeah, we talked about supporting this in the future, but for now I just hard coded it. If you do want to start building from other branches you may want to consider creating a branch of this script/branch for each of those other branches otherwise it will be hard to track down your build results because they will all be under this one job.